### ISSUE TITLE
Allow specifying a maximum amount of past back up multiple backups

### ISSUE THREAD

**Issue by Atrate** [MuntashirAkon/AppManager#221]
A useful feature would be to allow the user to set a maximum amount of backups to keep when using "back up multiple" so that it can be set in a profile and run e.g. every day without fear that it would cause too much storage to be used.

AM would then, on backup, decide if the next backup is going to be "too many" and if so, delete the oldest one from the app's backups folder.

Bonus feature: freezing backups — a single backup could be "frozen" or "locked" so that it is never deleted by AM.

**Comment by MuntashirAkon** [comment]
> AM would then, on backup, decide if the next backup is going to be "too many" and if so, delete the oldest one from the app's backups folder.

Yeah, this could be done. To improve efficiency, AM will only check this for the selected packages (this should be the logical way of handling them). But in settings, when is option will be set, AM will scan all backups and remove the old ones immediately.

> freezing backups — a single backup could be "frozen" or "locked" so that it is never deleted by AM.

In order to do it efficiently, AM will create a single empty file called `.freeze` inside the corresponding backup folder (ie. `package_name/backup_name`) which will restrict AM from deleting the backup.

**Comment by MuntashirAkon** [comment]
Since fb83787e7ede2db875676dccf1f7fb187e033af7, freezing/locking can be configured inside App Manager.

**Comment by username-is-not** [comment]
IMO this is too complex and unnecessary (time better spent elsewhere! 🙂); and for incremental backups and such, we can use different software such as the mature (fork of OAndBackup) Neo Backup application: on F-Droid and GitHub.

**Comment by MuntashirAkon** [comment]
> IMO this is too complex and unnecessary (time better spent elsewhere! 🙂); and for incremental backups and such, we can use different software such as the mature (fork of OAndBackup) Neo Backup application: on F-Droid and GitHub.

Thanks for your opinion. Perhaps you don't know: App Manager is a suite of tools unlike those software (remember what the description says _App Manager began by merging such and such apps_). So, if you apply this argument, App Manager should not exist in first place.
